app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java

```diff
@@ -46,78 +49,95 @@ public VideoPlaybackResolver(@NonNull final Context context,
-                    PlayerHelper.cacheKeyOf(info, video),
-                    MediaFormat.getSuffixById(video.getFormatId()), tag);
-            mediaSources.add(streamSource);
-        }
+            final List<MediaSource> mediaSources = new ArrayList<>();
+
+            // Create video stream source
+            final List<VideoStream> videos = ListHelper.getSortedStreamVideosList(context,
+                    info.getVideoStreams(), info.getVideoOnlyStreams(), false, true);
+            final int index;
+            if (videos.isEmpty()) {
+                index = -1;
+            } else if (playbackQuality == null) {
+                index = qualityResolver.getDefaultResolutionIndex(videos);
+            } else {
+                index = qualityResolver.getOverrideResolutionIndex(videos, getPlaybackQuality());
+            }
+            final MediaSourceTag tag = new MediaSourceTag(info, videos, index);
+            @Nullable final VideoStream video = tag.getSelectedVideoStream();
+
+            if (video != null) {
+                final MediaSource streamSource = buildMediaSource(dataSource, video.getUrl(),
+                        PlayerHelper.cacheKeyOf(info, video),
+                        MediaFormat.getSuffixById(video.getFormatId()), tag);
+                mediaSources.add(streamSource);
+            }
 
-        // Create optional audio stream source
-        final List<AudioStream> audioStreams = info.getAudioStreams();
-        final AudioStream audio = audioStreams.isEmpty() ? null : audioStreams.get(
-                ListHelper.getDefaultAudioFormat(context, audioStreams));
-        // Use the audio stream if there is no video stream, or
-        // Merge with audio stream in case if video does not contain audio
-        if (audio != null && (video == null || video.isVideoOnly)) {
-            final MediaSource audioSource = buildMediaSource(dataSource, audio.getUrl(),
-                    PlayerHelper.cacheKeyOf(info, audio),
-                    MediaFormat.getSuffixById(audio.getFormatId()), tag);
-            mediaSources.add(audioSource);
-        }
+            // Create optional audio stream source
+            final List<AudioStream> audioStreams = info.getAudioStreams();
+            final AudioStream audio = audioStreams.isEmpty() ? null : audioStreams.get(
+                    ListHelper.getDefaultAudioFormat(context, audioStreams));
+            // Use the audio stream if there is no video stream, or
+            // Merge with audio stream in case if video does not contain audio
+            if (audio != null && (video == null || video.isVideoOnly)) {
+                final MediaSource audioSource = buildMediaSource(dataSource, audio.getUrl(),
+                        PlayerHelper.cacheKeyOf(info, audio),
+                        MediaFormat.getSuffixById(audio.getFormatId()), tag);
+                mediaSources.add(audioSource);
+                isVideoAndAudioSeparated = true;
```

Comment:
~~Shouldn't this be `isVideoAndAudioSeparated = (video != null);`? Ok, the video is null only in corner cases, so the difference would never happen, but technically~~ I changed my mind, this is right as it is, since when there is no video audio is still separated from video